### DISCUSSION TITLE
Problem with SessionBagInterface "pimcore_admin" is not registered

### DISCUSSION THREAD

**Discussion by ziggurad** [symfony/symfony#51480] [Q&A]
Hello!

At first I thought my problem was related to the Pimcore framework. Now I don't know myself and I'm looking for help ;)

My project uses Pimcore framework which uses symfony. The most important libraries affected by the problem:
symfony/http-foundation: v5.4.25
friendsofsymfony/jsrouting-bundle: v2.8.0

There are 9 Session Storage Bags for all requests, only for request /js/routing?callback=fos.Router.setData i have only 2 bags and get error: "InvalidArgumentException: The SessionBagInterface "pimcore_admin" is not registered. in /var/www/pimcore/vendor/symfony/http-foundation/Session/Storage/NativeSessionStorage.php:347"

More info:
https://github.com/pimcore/pimcore/discussions/15805
https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/issues/418

Can anyone give any hint? I'm slowly thinking about giving up programming :D

**Comment by stof** [comment]
Symfony itself does not have 9 session bags. There are only 2 (the default one for normal session attributes and the one for flash messages)

My guess is that pimcore registers those bags only when some condition happens, while you have code that executes even when this condition is not met.
But without even having a stack trace to know where this bag tries to be accessed from the session, there is nothing we can do to help you.
